internal/usde/usde.go

```diff
@@ -54,11 +53,11 @@ func resourceSpecNeedsUpdating(ctx context.Context, newDef, existingDef *unstruc
 
 	// Get data loss fields config based on the spec type (Pipeline, ISBS)
 	dataLossFields := []config.SpecDataLossField{}
-	if reflect.DeepEqual(newDef.GroupVersionKind(), numaflowv1.PipelineGroupVersionKind) {
+	if util.CompareStructNumTypeAgnostic(newDef.GroupVersionKind(), numaflowv1.PipelineGroupVersionKind) {
 		dataLossFields = usdeConfig.PipelineSpecDataLossFields
-	} else if reflect.DeepEqual(newDef.GroupVersionKind(), numaflowv1.ISBGroupVersionKind) {
+	} else if util.CompareStructNumTypeAgnostic(newDef.GroupVersionKind(), numaflowv1.ISBGroupVersionKind) {
 		dataLossFields = usdeConfig.ISBServiceSpecDataLossFields
-	} else if reflect.DeepEqual(newDef.GroupVersionKind(), apiv1.NumaflowControllerGroupVersionKind) {
+	} else if util.CompareStructNumTypeAgnostic(newDef.GroupVersionKind(), apiv1.NumaflowControllerGroupVersionKind) {
```

Comment:
Sorry, I know it's not hurting anything to have this, but it may create more confusion to read. Maybe we should just put these 3 back to DeepEqual() function?

Reply:
(since there are no numbers involved)

Reply:
Sure, NP